runtime/src/accounts_db.rs

```diff
@@ -706,17 +706,26 @@ impl AccountsDB {
 
         let mut reclaims_time = Measure::start("reclaims");
         // Recalculate reclaims with new purge set
+        let purges_key_to_slot_set: Vec<_> = purges
+            .into_iter()
+            .map(|(key, (slots_list, _ref_count))| {
+                (
+                    key,
+                    HashSet::from_iter(slots_list.into_iter().map(|(slot, _)| slot)),
+                )
+            })
+            .collect();
+        let accounts_index = self.accounts_index.read().unwrap();
         let mut reclaims = Vec::new();
         let mut dead_keys = Vec::new();
-        for pubkey in purges.keys() {
-            let (new_reclaims, is_empty) = accounts_index.purge(&pubkey);
+        for (pubkey, slots_set) in purges_key_to_slot_set {
```

Comment:
[then](https://github.com/solana-labs/solana/pull/10898/files?diff=unified&w=1#r449371680), this nicely would read like this:

`for (pubkey, slot_set) in pubkey_to_slot_set {`